app/controllers/discourse_automation/automations_controller.rb

```diff
@@ -0,0 +1,13 @@
+# frozen_string_literal: true
+
+module DiscourseAutomation
+  class AutomationsController < ApplicationController
+    requires_login
+
+    def trigger
```

Comment:
I am confused why we have this separate route to the one in app/controllers/admin/discourse_automation/admin_discourse_automation_automations_controller.rb when this one has an admin constraint as well?

Reply:
Also I am not sure if this is what we want, but I can just pass any params to this and it gives me a 200 OK response. Should there be parameter validation based on the automation script? I think I mentioned this kind of thing in my feedback too. It's hard to tell whether the trigger actually did anything or not without debugging in rails.

Reply:
> I am confused why we have this separate route to the one in app/controllers/admin/discourse_automation/admin_discourse_automation_automations_controller.rb when this one has an admin constraint as well?

it's explained in the commit message, I wanted a simpler one. /admin/plugins/discourse-automation/... seems too complicated

Reply:
> Also I am not sure if this is what we want, but I can just pass any params to this and it gives me a 200 OK response. Should there be parameter validation based on the automation script? I think I mentioned this kind of thing in my feedback too. It's hard to tell whether the trigger actually did anything or not without debugging in rails.

Why though ? I have added params slicing at the beginning which means you could say I want only foo and bar and not  baz in the payload, but what does it give us?

I think it's script responsibility to validate params, I can't do this at an higher level. If I do this is going to be ultra limited, I don't want to reinvent a whole syntax to validate deep nested objects

Reply:
> I think it's script responsibility to validate params

Yeah I think this makes sense; maybe a script can raise an error if it doesn't have certain required params?

Reply:
> it's explained in the commit message, I wanted a simpler one. /admin/plugins/discourse-automation/... seems too complicated

Yes sorry I realise that, I mean that you have left the old one in place too; why not just delete the old one at the same time?